### ISSUE TITLE
Implement PIN 6

### ISSUE THREAD

**Issue by cicdw** [PrefectHQ/prefect#915]
https://docs.prefect.io/guide/PINs/PIN-6-Remove-Constant-Tasks.html

Basically, to remove the auto-generation of "Constant" task and instead store constants in the flow object.

**Comment by cicdw** [comment]
@jlowin I have a different proposal for how to implement this, based on mapping over a constant list:
```python
add_one.map(x=[1, 2, 3])
```
With this use case in mind, it's easier to implement a special `Edge` property called `upstream_constant` or something, which tracks whether the upstream task is a constant or not.  In fact, we could compute this on the fly by checking whether the upstream is a Task instance or not.  Alternatively, a new initialization kwarg which holds an `upstream_constant` instead of an `upstream_task`.

Implemented this way, the information is still contained entirely within the Flow, but it also allows for edges such as
```python
Edge(upstream_constant=[1, 2, 3], downstream_task=add_one, key='x', mapped=True)
```
which handle mapping over constants.

**Comment by jlowin** [comment]
Following offline conversation in which we realized that any attempt to modify Edge objects would essentially reimplement Constant Tasks by another name, we've decided to decline PIN 6 (for now) -- See #917

**Comment by cicdw** [comment]
Closed via #915
